src/python/WMCore/Storage/RucioFileCatalog.py

```diff
@@ -0,0 +1,236 @@
+        """
+        _matchLFN_
+
+        Return the result for the LFN provided if the LFN
+        matches the path-match for that protocol
+
+        Return None if no match
+
+        """
+        result = self._doMatch(protocol, lfn, "lfn-to-pfn", self.matchLFN)
+        return result
+
+    def matchPFN(self, protocol, pfn):
+        """
+        _matchLFN_
+
+        Return the result for the LFN provided if the LFN
+        matches the path-match for that protocol
+
+        Return None if no match
+
+        """
+        result = self._doMatch(protocol, pfn, "pfn-to-lfn", self.matchPFN)
+        return result
+    def __str__(self):
+        result = ""
+        for mapping in ['lfn-to-pfn', 'pfn-to-lfn']:
+            for item in self[mapping]:
+                result += "\t%s: protocol=%s path-match-re=%s result=%s" % (
+                    mapping,
+                    item['protocol'],
+                    item['path-match-expr'].pattern,
+                    item['result'])
+                if item['chain'] != None:
+                    result += " chain=%s" % item['chain']
+                result += "\n"
+        return result
+
+
+def storageJsonPath(currentSite, currentSubsite, storageSite):
+    """
+    Return a path to storage.json from site names
+    :para currentSite and currentSubsite are the names of facilities where jobs will be executed, storageSite is the storage facility for a stage-out
+    :return /pathToStorageJson/storage.json
+    """
+    #get site config
+    siteConfigPath = os.getenv('SITECONFIG_PATH',None)
+    if not siteConfigPath:
+        raise RuntimeError('SITECONFIG_PATH is not defined')
+    subPath = ''
+    #the storage site is where jobs are executed so use local path given in SITECONFIG_PATH to locate storage.json
+    if currentSite == storageSite:
+        #it is a site (no defined subSite), storage.json is located at the path given in SITECONFIG_PATH
+        if currentSubsite is None:
+            subPath = siteConfigPath
+        #it is a subsite, move one level up
+        else:
+            subPath = siteConfigPath + '/..'
+    #cross site
+    else:
+        #it is a site (no defined subSite), move one level up
+        if currentSubsite is None:
+            subPath = siteConfigPath + '/../' + storageSite
+        #it is a subsite, move two levels up
+        else:
+            subPath = siteConfigPath + '/../../' + storageSite
+    pathToStorageDescription = subPath + '/storage.json'
+    pathToStorageDescription = os.path.normpath(os.path.realpath(pathToStorageDescription))#resolve symbolic link and relative path?
+    return pathToStorageDescription 
+
+def readRFC(filename, storageSite, volume, protocol):
+    """
+    _readRFC_
+
+    Read the provided storage.json and return a RucioFileCatalog
+    instance containing the details found in it
+    """
+    rfcInstance = RucioFileCatalog()
+    try:
+        jsonFile = open(filename,encoding="utf-8")
```

Comment:
We better use context manager's for opening files to avoid leaking file descriptors.

Reply:
Done